### PR TITLE
gh-105724: Add location information to `assert` errors

### PR DESCRIPTION
**Pull request by sobolevn**
New error message for `assert 1 > 2` is:

```python
Traceback (most recent call last):
  File "/Users/sobolev/Desktop/cpython/ex.py", line 1, in <module>
    assert 1 > 2
           ^^^^^
AssertionError     
```

I've also included test cases where `assert` takes multiple lines. 

* Issue: gh-105724

### CODE REVIEW

**Review by markshannon — APPROVED** [review]
Nice. Thanks.

**Comment by markshannon** [comment]
`test_multiline_assert` in `test_compile` needs to be updated, otherwise LGTM.

File: Lib/test/test_exceptions.py [hunk 1]
```diff
@@ -1931,6 +1931,123 @@ def test_copy_pickle(self):
...
+            ('assert 1 > 2',
+                [
+                    '    assert 1 > 2',
+                    '           ^^^^^',
+                    'AssertionError',
+                ],
+            ),
+            ('assert 1 > 2 and 3 > 2',
+                [
+                    '    assert 1 > 2 and 3 > 2',
```
Comment (iritkatriel) [inline Lib/test/test_exceptions.py:1970]: Would be nice if it was possible to show which of these has failed.

Comment (markshannon) [reply Lib/test/test_exceptions.py:1970]: It would be. But that's for another PR, IMO.
I'll elaborate on the issue.

**Pull request merged**